src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -118,6 +118,25 @@ export class NotebookLinkHandler {
 		}
 	}
 
+	public getEncodedLinkUrl(): string | undefined {
+		// Need to encode URI here in order for user to click the proper encoded link in WYSIWYG
+		// skip encoding it if it's already encoded
+		if (typeof this._link === 'string') {
+			if (!this.containsEncodedComponents(this._link)) {
+				return encodeURI(this._link);
+			}
+			return this._link;
+		}
+		return undefined;
+	}
+
+	// returns true if the passed in url has encoded strings:
+	// space gets encoded as %20 and ;,/?:@&=+$ -> %3B%2C%2F%3F%3A%40%26%3D%2B%24
+	containsEncodedComponents(url): boolean {
```

Comment:
1. This should still be outside the class - and TBH I'd suggest make a `src/sql/base/common/network.ts` file and putting it there since it's a pretty generic thing
2. Please use valid JS doc (/** ... */) comments
3. I'm not really understanding the second part - this isn't actually encoding anything so not sure how it matters. You should be defining what an "encoded string" is and a quick description of how it detects that.